packages/providers/src/transaction-request/transaction-request.ts

```diff
@@ -336,12 +341,19 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
 
   /**
    * Adds a single resource to the transaction by adding a coin/message input and a
-   * change output for the related assetId, if one it was not added yet.
+   * change output for the related assetId or predicate, if one of it was not added yet.
    *
    * @param resource - The resource to add.
    * @returns This transaction.
    */
-  addResource(resource: Resource) {
+  addResource<T extends AbstractAccount>(resource: AccountResource<T>) {
+    const account = resource.account;
+
+    if ((account as AbstractPredicate).bytes) {
```

Comment:
It's not possible to use `account instanceof AbstractPredicate` because `Predicate` itself just implements `AbstractPredicate`, it doesn't extend it; it extends `Account`.